test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,328 @@
+        std::vector<alphabet_t> read_tmp{std::ranges::begin(ref) + rpos,
+                                         std::ranges::begin(ref) + rpos + read_length + simulated_errors};
+
+        for (uint8_t j = 0; j < simulated_errors; ++j)
+        {
+            double prob = probability(gen);
+            // Substitution
+            if (prob_insertion + prob_deletion < prob)
+            {
+                mutate_substitution(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Insertion
+            else if (prob_insertion < prob)
+            {
+                mutate_insertion(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Deletion
+            else
+            {
+                mutate_deletion(read_tmp, simulated_errors, seeds(gen));
+            }
+        }
+        read_tmp.erase(std::ranges::begin(read_tmp) + read_length, std::ranges::end(read_tmp));
+        reads.push_back(read_tmp);
+    }
+    return reads;
+}
+
+template <typename alphabet_t>
+std::vector<alphabet_t> generate_repeating_sequence(size_t const template_length = 5000,
+                                                    size_t const repeats = 20,
+                                                    double const template_fraction = 1,
+                                                    size_t const seed = 0)
+{
+    std::vector<alphabet_t> seq_template = generate_sequence<alphabet_t>(template_length, 0, seed);
+
+    // copy substrings of length len from seq_template mutate and concatenate them
+    size_t len = std::round(template_length * template_fraction);
+    uint8_t simulated_errors = 5;
+    len = (len + simulated_errors  > template_length) ? template_length - simulated_errors : len;
+
+    return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15) | view::persist | std::view::join;
+}
+
+//============================================================================
+//  undirectional; trivial_search, collection, dna4, all-mapping
+//============================================================================
+
+void unidirectional_search_all_collection(benchmark::State & state, options && o)
+{
+    size_t set_size = 10;
+    std::vector<std::vector<seqan3::dna4>> collection;
+    std::vector<std::vector<seqan3::dna4>> reads;
+    for (size_t i = 0; i < set_size; ++i)
+    {
+        collection.push_back(generate_sequence<seqan3::dna4>(o.sequence_length, 0, 0));
+        std::vector<std::vector<seqan3::dna4>> seq_reads = generate_reads(collection.back(), o.number_of_reads,
+                                                                          o.read_length, o.simulated_errors,
+                                                                          o.prob_insertion, o.prob_deletion,
+                                                                          o.stddev);
+        std::move(std::ranges::begin(seq_reads), std::ranges::end(seq_reads), std::back_inserter(reads));
```

Comment:
What happens here? :thinking:

Reply:
Well I hope this happens: 
reads.insert(std::ranges::end(reads), std::ranges::begin(seq_reads), std::ranges::end(seq_reads));
But instead of copying we move the elements of seq_reads into reads. 
Overall reads create from each sequence in the collection are merge in to single vector.

Reply:
I still don't understand this. `std::move` takes exactly one argument and the effects of std::move are exactly:
```
static_cast<typename std::remove_reference<T>::type&&>(t) 
```
See also https://en.cppreference.com/w/cpp/utility/move

Reply:
It's algorithm/move, not utility/move. https://en.cppreference.com/w/cpp/algorithm/move

Reply:
:scream: What's dis??

Unbelievable how they have two function of the same name that do entirely different things...

Anyway, in that case, please use std::ranges::move instead!

Reply:
for `std::ranges::move` you can use `seq_reads` straight away and don't need to do the begin, end - thing.

Reply:
Ok, done.